### PR TITLE
SRCH-855-remove unused columns from users table

### PR DESCRIPTION
**Pull request by peggles2**
This pull request includes the following:
Removed following columns from user table
password_updated_at
perishable_token
crypted_password
password_salt
failed_login_count
email_verification_token

Update rspec tests
remove complete_registration test 
complete registration behavior is for a user to verify their emails that is handled by login.gov.  In the future something similar will be developed to direct the user to complete their information. Removed it for now.

removed cucumber tests
updated fixtures.

SRCH-1127 - cleanup any authentication reference in code has also been implemented in this pull request.

**DEPLOYMENT NOTES:** 
email_templates needs to be reloaded after deployment.

### CODE REVIEW

File: spec/fixtures/users.yml [hunk 1]
```diff
@@ -31,13 +37,23 @@ affiliate_manager:
   contact_name: Affiliate Manager
   welcome_email_sent: true
 
+affiliate_manager_with_pending_email_verification_status:
```
Comment (peggles2) [inline spec/fixtures/users.yml]: Tried to rename but started to become a yak shaving processes so leaving it here for now.

**Comment by peggles2** [comment]
The info re. email verification in the README - I saw that before and did not want to touch it in this story.

File: db/migrate/20200108160640_remove_user_unused_columns.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class RemoveUserUnusedColumns < ActiveRecord::Migration[5.2]
+  def change
```
[the remark this answers is not in the document]
Comment (peggles2) [reply db/migrate/20200108160640_remove_user_unused_columns.rb]: that is going to be addressed in https://cm-jira.usa.gov/browse/SRCH-1219

File: spec/models/user_spec.rb [hunk 3]
```diff
@@ -21,14 +21,18 @@
...
 
   describe 'schema' do
-    it { is_expected.to have_db_column(:failed_login_count).of_type(:integer).with_options(default: 0, null: false) }
-    it { is_expected.to have_db_column(:password_updated_at).of_type(:datetime).with_options(null: true) }
+    it { is_expected.not_to have_db_column(:failed_login_count) }
+    it { is_expected.not_to have_db_column(:password_updated_at) }
+    it { is_expected.not_to have_db_column(:perishable_token) }
+    it { is_expected.not_to have_db_column(:crypted_password) }
+    it { is_expected.not_to have_db_column(:failed_login_count) }
+    it { is_expected.not_to have_db_column(:email_verification_token) }
```
[the remark this answers is not in the document]
Comment (peggles2) [reply spec/models/user_spec.rb]: yeah either way is fine.  It was mostly to test the changes.

**Comment by MothOnMars** [comment]
A couple more changes: 
- The specs related to SRCH-953 still need to be removed.
- Something I noticed this time that I didn't last time is that there is still quite a bit of cruft related to email verification. We should clean that up:
```
$ git grep email_verification | grep -v db/migrate
README.markdown:Look for the `email_verification_token` in your rails server stdout and open the verification link in your favorite browser:
README.markdown:    http://127.0.0.1:3000/email_verification/<email_verification_token>
app/helpers/sites_helper.rb:                when 'pending_email_verification', 'pending_approval'
app/models/email_template.rb:      user_email_verification: "[Search.gov] Verify your email",
config/routes.rb:  resources :email_verification, only: :show
db/email_templates/user_email_verification.html.erb:                                <a class="mcnButton " title="Verify and log in" href="<%= @email_verification_url %>" target="_blank" style="font-weight: normal;letter-spacing: normal;line-height: 100%;text-align: center;text-decoration: none;color: #FFFFFF;">Verify and log in</a>
db/email_templates/user_email_verification.html.erb:<span style="color:#0088cc"><span style="line-height:20.8px"><%= @email_verification_url %></span></span><br style="line-height: 20.8px;">
features/users.feature:    And I should see "https://localhost:3000/email_verification" in the email body
spec/helpers/sites_helper_spec.rb:    context 'when User has', approval_status: 'pending_email_verification' do
spec/mailers/emailer_spec.rb:    let(:user) { double(User, :email => "invitee@agency.com", :contact_name => 'Invitee Joe', :email_verification_token => 'some_special_token', affiliates: []) }
spec/models/user_spec.rb:      before { allow(Emailer).to receive(:user_email_verification).with(an_instance_of(User)).and_return @emailer }
spec/models/user_spec.rb:          .from('approved').to('pending_email_verification')
spec/models/user_spec.rb:        expect(Emailer).to receive(:user_email_verification).with(user)
spec/models/user_spec.rb:        @user = users(:affiliate_added_by_another_affiliate_with_pending_email_verification_status)
spec/models/user_spec.rb:        expect(@user.is_pending_email_verification?).to be true
spec/models/user_spec.rb:        expect(@user.is_pending_email_verification?).to be true
spec/models/user_spec.rb:          user.update(approval_status: 'pending_email_verification',
spec/models/user_spec.rb:        expect(Emailer).to_not receive(:new_user_email_verification)
```

**Comment by peggles2** [comment]
Most of these were in pending or @wip tests. Meaning tests that were commented out to be addressed later.  I will cleanup a little but was originally planning to address when I worked on those Jira tickets.

**Comment by peggles2** [comment]
Did some more cleanup. Some commented out tests I left alone because they are going to be addressed when that part of the code is re-factored. I like to keep the test to tell me how it bahaved so I can mimic the behavior as closely as possible.

**Comment by peggles2** [comment]
closed this to separate the migrations from code. See https://github.com/GSA/search-gov/pull/518

**Pull request closed without merging**